pulsar-metadata/src/main/java/org/apache/pulsar/metadata/coordination/impl/ResourceLockImpl.java

```diff
@@ -71,7 +73,14 @@ public synchronized T getValue() {
 
     @Override
     public synchronized CompletableFuture<Void> updateValue(T newValue) {
-       return acquire(newValue);
+        if (state == State.Revalidating) {
+            return revalidateFuture
+                    // ignore revalidate result
+                    .exceptionally(ignore -> null)
+                    .thenCompose(__ -> acquire(newValue));
```

Comment:
Is it possible to add a test to cover this case?

Reply:
I can try on it.